pkg/cloud/aws/actuators/cluster/actuator.go

```diff
@@ -122,9 +132,56 @@ func (a *Actuator) Reconcile(cluster *clusterv1.Cluster) error {
 		return errors.Wrapf(err, "failed to get kubeconfig secret for cluster %q", cluster.Name)
 	}
 
+
+
+	if cluster.Annotations[v1alpha1.AnnotationControlPlaneReady] != "true" {
+		log.Info("Cluster does not have ready annotation - checking for ready control plane machines")
+
+		machines, err := a.client.Machines(cluster.Namespace).List(actuators.ListOptionsForCluster(cluster.Name))
+		if err != nil {
+			return errors.Wrapf(err, "failed to list machines for cluster %q", cluster.Name)
+		}
+
+		controlPlaneMachines := machine.GetControlPlaneMachines(machines)
+
+		machineReady := false
+		for _, machine := range controlPlaneMachines {
+			if machine.Status.NodeRef != nil {
```

Comment:
I can't think of another way to do this check, but IIRC the node controller doesn't work for "management cluster" style topologies. It seems like there'd be a workaround in manually flagging the control plane as "ready", should we  document and/or tool that flow?

Reply:
Since we know we have a load balancer, and the first api endpoint is that load balancer, and we know we have credentials for the cluster, could we not just query the /healthz endpoint for the api endpoint?

Reply:
The NodeRef controller works for all clusters, management or otherwise. This is new in CAPI v0.1.4 and CAPA has been updated to work with it.

Reply:
@detiber we could, but that requires an extra HTTP call - do you want me to add that? I figured this was a pretty clean way to determine things.

Reply:
The NodeRef controller wouldn't work for a pod-based control plane, or an externally managed controlplane. Or would we just say that those use cases need to set the annotation manually?

Reply:
Good points. How about a TODO for now to consider /healthz?

Reply:
> The NodeRef controller wouldn't work for a pod-based control plane

I consider those edge cases. We're kind of all in on the NodeRef at this point. If there's an exception, I feel like another controller that sets some other version of a NodeRef/OwnerRef/RelatedRef should exist?

Reply:
They may very well be edge cases, but we do need to document how those cases could be handled, especially since we know there are people using cluster-api for this today.

Reply:
@detiber given that this code is in CAPA, and CAPA only supports machine-based control plane members today, do you still think there is anything actionable here?

Reply:
@ncdc I think we still need to document it, since someone potentially could be using CAPA with a pre-existing external control plane. Otherwise they would be prevented from creating any Machines.

Reply:
Fair point. Maybe it makes more sense to switch to the /healthz check? Or would that have edge cases too?

Reply:
I think just documenting the annotations should be sufficient, that should cover the external use cases.

Reply:
I thought we were talking about NodeRef?

Reply:
We are, but setting of the annotation shortcuts the handling of NodeRef for the control plane and unblocks Machine joins.